app/modules/intelligence/agents/custom_agents/custom_agent_router.py

```
@@ -0,0 +1,119 @@
+        )
+    except HTTPException as he:
+        raise he
+    except Exception as e:
+        logger.error(f"Error updating custom agent: {str(e)}")
+        raise HTTPException(status_code=500, detail="Internal server error")
+
+
+@router.get("/{agent_id}", response_model=Agent)
+async def get_custom_agent_info(
+    agent_id: str,
+    db: Session = Depends(get_db),
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)